tests/test_utils.py

```diff
@@ -50,6 +52,45 @@ def eos_id(self):
     def bos_id(self):
         return 0
 
+    def tokenize_messages(
```

Comment:
We should think of an alternative solution to this, because we'll have to update this every time the real tokenize messages is updated. Can we do a more stripped down approach for testing purposes or is this the most barebones?

Reply:
Yeah I agree. The main thing I wanted here was to guarantee I could replicate the performance in the existing unit test using the same logic. We can definitely use a simpler method but will have to change the expected values. (Really I should just add a test for `tokenize_messages` on the tokenizer, then we can use something simple here and still be confident it's working as expected.)

Reply:
you added the `tokenize_messages` test - should we simplify here?

Reply:
Yeah, it is nice to explicitly test the tokenizer on the chat dataset though. I have a better idea (spoiler alert: it involves inheritance 😃)

Reply:
Yeah, I do like explicitly testing fetching a sample from ChatDataset though. I have another idea to simplify the code here 😃